### ISSUE TITLE
Meshcentral broken,  reinstall needed

### ISSUE THREAD

**Issue by merlinweb** [Ylianst/MeshCentral#244]
I have a strange issue, started yesterday morning, happened this morning too.

I've found MeshCentral unoperational, service halted, in mesherrors I've found following lines repeating every 5 seconds:
```
-------- 2019-5-23 00:56:58 ---- 0.3.5-a --------

module.js:545
throw err;
^

Error: Cannot find module 'C:\Program Files\Open Source\MeshCentral\node_modules\meshcentral\winservice.js'
at Function.Module._resolveFilename (module.js:543:15)
at Function.Module._load (module.js:470:25)
at Function.Module.runMain (module.js:690:10)
at startup (bootstrap_node.js:194:16)
at bootstrap_node.js:666:3

```
Folder "C:\Program Files\Open Source\MeshCentral\node_modules\meshcentral" is completely Empty, except "Daemon", the only folder present

Solved with a reinstall and restoring backupped files

I'm afraid it will happen tomorrow

Rebooting server (a VMWare virtual machine windows Server 2016), won't cause any problem

Any help is very appreciated

Thanks

Merlinweb

**Comment by PathfinderNetworks** [comment]
I had exactly the same issue today!  Although, in my case, the only folder missing from the C:\Program Files\Open Source\MeshCentral\node_modules\meshcentral was the daemon folder.  I had seen this happen before when trying to use the update feature through the website GUI.  So I wonder if something triggered an automatic version upgrade even though it shouldn't?

**Comment by merlinweb** [comment]
Thanks, looking at my config.json, I have selfupdate enabled, actually
I've disabled auto update, I will check tomorrow morning to verify server (it's 6.00 PM here...)

Cheers

Merlinweb

**Comment by Ylianst** [comment]
Thanks for the reports. I run all my servers on Linux and so, would not see these problems. Probably not a good idea to enable "selfupdate" on any important server, I don't do it on my servers since I want to be present when an update occurs. Updating from the "My Server" tab is quick and easy.

I will do some testing on Windows now.

**Comment by merlinweb** [comment]
Honestly, the problem of deleted files only appeared since yesterday, but I've never had the auto-update enabled, if I remember correctly
I've always only used the Windows installation, manually after looking at github to see changes

**Comment by merlinweb** [comment]
No, definitely not a good idea.
I was sure it was disabled but, checking my backup, I was wrong...

**Comment by Ylianst** [comment]
If I had to guess, the problem is in the "package.json". The dependencies list has changed and probably caused the update to fail. Two notable changes is that I removed unsafe WebAuthn library and built my own and changed from "mongojs" to "mongodb" which is the official driver for MongoDB.

If I am right, just reinstall the latest version and it will work just fine. Make sure to backup "meshcentral-data", it's super important you keep a backup of that folder.

All this said, I will do some testing on Windows. It may be that auto-update on Windows when running as a service is a bad idea.

**Comment by Ylianst** [comment]
FYI. If you are running MeshCentral in production, keep a known good copy of the entire "node_modules" folder. If anything goes wrong, just stop the server and copy that folder back and start again.

**Comment by merlinweb** [comment]
uhm, yesterday I made a backup, uninstall everything without keeping files, done a fresh install and restoring data.
Today folders was deleted again.
now I've double checked there is no self update and  I am waiting for tomorrow morning... but I'm afraid bug is still there

**Comment by merlinweb** [comment]
> FYI. If you are running MeshCentral in production, keep a known good copy of the entire "node_modules" folder. If anything goes wrong, just stop the server and copy that folder back and start again.

just done this morning, after second deletion, but thanks

**Comment by PathfinderNetworks** [comment]
My MeshCentral install did the same thing merlinweb is discussing today after I had restored 0.3.4-y. Within a few hours the server quit working and, upon investigating, the only folder I had in node-modules/meshcentral was the daemon folder.  I restored 0.3.4-y again and checked my config.json. I, too, had the selfupdate set to true- even though I'm certain I hadn't set that before (my server has never automatically updated in the past).  I changed it to false and, so far, it's stayed running.
